### DISCUSSION TITLE
Is On:event supported?

### DISCUSSION THREAD

**Discussion by leejoramo-d51** [yesvelte/yesvelte#245] [Q&A]
In [Sveltestrap](https://github.com/bestguy/sveltestrap) I use on:blur on:change and the like on my forms. For example, every time blur is triggered on an input field, I save the value to a Store which also triggers validations. 

This doesn't seem to work in my testing of yesvelte. Here is an example of the type of think I have done in Sveltestrap. 

```html
<script>
  const formValues = store.formValues
  const handleInput = (e) => {    
    $formValues[name] = e.target.value
    formValues.update((v) => v)
  }    
</script>

<Input
  {id}
  {name}
  bind:value
  on:blur={handleInput}
  {type}
/ >

```

**Comment by pournasserian** [comment]
We were waiting for Svelte team to see if they will support wildcard event forwarding or not. This issue is opened since 2019:
https://github.com/sveltejs/svelte/issues/2837

We have developed a custom solution in today's version.

Would you update your package and let me know if the issue is resolved?

**Comment by leejoramo-d51** [comment]
I tried it with `Release 0.0.2-next.70` and `71` but it didn't work for me.

Do you have a simple example of events working?  Thanks

**Reply by TheHadiAhmadi** [reply]
@leejoramo-d51 sorry for that, in `Release 0.0.2-next.72` we added event forwarding for Form related components.

now you can use Form components with standard DOM Events.

```svelte
<Input on:blur={onBlur} on:change={onChange} on:click={onClick} />
<FormInput on:blur={onBlur} on:change={onChange} on:click={onClick} />
...
```

thanks

**Comment by pournasserian** [comment]
Here is a simple example:

```
<script>	
	import { Input } from 'yesvelte@next'
	let value
	 const handleInput = () => { 
		 console.log(value)
  }    
</script>

<svelte:head>
	<link rel="stylesheet" href="https://unpkg.com/yesvelte@next/css/tabler.min.css" />
</svelte:head>

<Input
  bind:value
  on:blur={handleInput}
/>
```

**Reply by leejoramo-d51** [reply]
I finally had time to try all of this. Your example works in a clean svelteKit install. 

However, when I try to do this within my project by replacing the Sveltestrap components with Yesvelte components, the `on:blur` event is not triggered.

`on:blur`  works fine in Sveltestrap, and the rest of Yesvelte is working properly. 

I will dig deeper to figure out what is different. My project is much larger than just this integration so I will need to create a minimal reproduction of the issue to share with with you.
